File: Spring Boot version/src/main/java/Thalassa/Models/Block.java
package Thalassa.Models;

import com.arangodb.springframework.annotation.Document;
import com.arangodb.springframework.annotation.Field;
import org.springframework.data.annotation.Id;
import java.math.BigInteger;
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;

@Document("Blockchain")
public class Block {

    @Id
    private String index;
    @Field("Hash")
    private String hash;
    @Field("Timestamp")
    private long timestamp;
    @Field("Data")
    private String data;
    @Field("Previous hash")
    private String previousHash;

    public Block() {
    }

    public Block(String index, String data, String previousHash) throws Exception {
        this.index = index;
        this.timestamp = System.currentTimeMillis();
        this.data = data;
        this.previousHash = previousHash;
        this.hash = calculateHash();
    }

    public Block(String index, String data) throws Exception {
        this.index = index;
        this.timestamp = System.currentTimeMillis();
        this.data = data;
        this.hash = calculateHash();
    }

    public String getIndex() {
        return index;
    }

    public String getHash() {
        return hash;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public String getPreviousHash() {
        return previousHash;
    }

    public String calculateHash() throws Exception {
        String blockData = previousHash == null ? index + timestamp + data : index + timestamp + data + previousHash;
        return new BigInteger(1, MessageDigest.getInstance("SHA-256").digest(blockData.getBytes(StandardCharsets.UTF_8))).toString(16);
    }
}

File: ASP.NET Core version/Models/Block.cs
using System;
using System.Security.Cryptography;
using System.Text;
using ArangoDB.Client;
using Newtonsoft.Json;

namespace Thalassa.Models
{
    [CollectionProperty(CollectionName = "Blockchain")]
    public class Block
    {
        [DocumentProperty(Identifier = IdentifierType.Key)]
        public string Index { get; set; }
        
        [DocumentProperty(PropertyName = "hash")]
        public string Hash { get; set; }
        
        [DocumentProperty(PropertyName = "timestamp")]
        public long Timestamp { get; set; }
        
        [DocumentProperty(PropertyName = "data")]
        public string Data { get; set; }
        
        [DocumentProperty(PropertyName = "previousHash"), JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public string PreviousHash { get; set; }

        [JsonConstructor]
        public Block() {}
        
        public Block(string index, string data, string previousHash)
        {
            Index = index;
            Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
            Data = data;
            PreviousHash = previousHash;
            Hash = CalculateHash();
        }

        public Block(string index, string data)
        {
            Index = index;
            Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
            Data = data;
            Hash = CalculateHash();
        }

        public string CalculateHash()
        {
            var blockData = Index + Timestamp + Data + PreviousHash;
            var result = new SHA256CryptoServiceProvider().ComputeHash(Encoding.UTF8.GetBytes(blockData));
            var sb = new StringBuilder();
            foreach (var t in result)
                sb.Append(t.ToString("x2"));

            return sb.ToString();
        }
    }
}